src/Arcus.WebApi.Logging.Core/Correlation/HttpCorrelationMessageHandler.cs

```diff
@@ -81,9 +81,9 @@ private CorrelationInfo DetermineCorrelationInfo()
             if (correlation is null)
             {
                 throw new InvalidOperationException(
-                    "Cannot enrich the HTTP request with HTTP correlation because no HTTP correlation was registered in the application, " +
-                    "make sure that you register the HTTP correlation services with 'services.WithHttpCorrelationTracking()' " +
-                    "and that you use the HTTP correlation middleware 'app.UseHttpCorrelation()' in API scenario's");
+                    "Cannot enrich the HTTP request with HTTP correlation because no HTTP correlation was registered in the application, " 
+                    + "make sure that you register the HTTP correlation services with 'services.AddHttpCorrelation()' " 
```

Comment:
> Here, you say that you should register http correlation by calling AddHttpCorrelation but in the test, we're still using WithHttpCorrelationTracking.

That's correct. The `WithHttpCorrelationTracking` will make use of the `IHttpCorrelationInfoAccessor` that will get registered by the `services.AddHttpCorrelation` call. The exception message was wrong here, so that's why the change.

> What is this WithHttpCorrelationTracking method used for ?

It was introduced in a previous PR: https://github.com/arcus-azure/arcus.webapi/pull/346 It made the HTTP dependency tracking more easily to lean on the built-in `IHttpClientFactory`.